### PR TITLE
Fab edits only

### PR DESCRIPTION
**Pull request by NZNightwolf**
Fabrication Additions
New Structures, New Recipes, New unlockable Recipes
Fabrication Guild updated

### CODE REVIEW

**Comment by NZNightwolf** [comment]
I'm happy to provide a list of all changes, as I know what every change/update in this is :)

Hoping to get this included in the next update, with the space stuff!

**Comment by NZNightwolf** [comment]
As Requested:

Toolset/Module Additions:
Generic Additions:
35 new Standard fabrication recipes.
25 new Fabrication guild recipes - Unlocked via guild tiers.
16 new Quest fabrication recipes.
Small area added to Veles Sewers for Quest giver.
NPC's added to Mon Cala Interior, CZ220 Hangar, Veles Sewers, Hutlar Outpost and Viscara Shop Interiors - Quest givers
Jedi Temple interior edited - At request of knights for IC/IG meetings

Items:
11 new Schematics/Patterns - Fabrication Recipes used to unlock recipes via quests/guild levels
1 new outfit - Nurse outfit for quest giver
77 placeable items - 76 from new fabrication recipes, 1 adds the single width panels wall (Which was missing)

Conversations:
6 new conversations for quest givers

Creatures:
4 new creatures - Quest givers
2 creature edits - Byysk guardians variables edited to give Quest Grouping ID.

Shops:
9 shops edited - Fabrication guild shop: New recipe unlocks added, and weight reduction bags. Agriculture guild shop: Weight reduction bags

Placeables:
3 new placeables - Edits of original placeables to remove useability and inventory

Code:
Quest Definition:
5 planets have new quests added - 1x in Hutlar, Tattooine, CZ220, Moncala, 2x in Viscara

Furniture Recipes:
77 new recipes added to list - 76 from new content, 1 from missing single width panels wall

Recipe Type:
As above

NPC Group Types:
Byysk guardians added to new Quest ID Type for Hutlar Quest.

Structure Type:
Added all 77 new structures. Filled in gaps so list is complete to #329

**Comment by taelon875** [comment]
Looks clear to me.

**Review by zunath — APPROVED** [review]

**Review by zunath — CHANGES_REQUESTED** [review]
Realizing a bunch of merge problems. Please review & correct, then I'll merge it in.

**Comment by NZNightwolf** [comment]
Hey Zunath,

Have done the conflicts you've requested, and it looks good on my end. Can you take a look when you get a second :)

**Review by NZNightwolf — COMMENTED** [review]
This all looks like what I submitted.

File: SWLOR.Game.Server/Feature/QuestDefinition/CZ220QuestDefinition.cs [hunk 1]
```diff
@@ -253,5 +254,21 @@ private static void TheColicoidExperiment(QuestBuilder builder)
...
+        private static void ScrapMetalMonstrosity(QuestBuilder builder)
+        {
+            builder.Create("scrapmetal_monster", "Scrap Metal Monstrosity")
+
+                .AddState()
+                .SetStateJournalText("5CR4P seems like they would like some more scrap metal. Maybe you should gather a lot for them.")
+                .AddCollectItemObjective("scrap_metal", 150)
+
+                .AddState()
+                .SetStateJournalText("That's a lot of scrap metal, hopefully it's enough. See 5CR4P has to beep.")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/CZ220QuestDefinition.cs:267]: I'm confused by the last sentence: "See 5SR4P has to beep." - Is this typo'd?

File: SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs [hunk 2]
```diff
@@ -155,5 +156,42 @@ private void StupendousSlugBile()
                 .AddItemReward("slug_shake", 1);
         }
 
+        private void BreakTheByysk()
+        {
+            _builder.Create("break_the_byysk", "Break the Byysk")
+
+                .AddState()
+                .SetStateJournalText("Sharene wants you to kill One Hundred Byysk. Of you go!")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs:164]: Bunch of grammar issues here.

"Sharene wants you to kill one hundred Byysk. Off you go!"

File: SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs [hunk 3]
```diff
@@ -155,5 +156,42 @@ private void StupendousSlugBile()
...
+                .AddKillObjective(NPCGroupType.Byysk_Guardian, 100)
+
+                .AddState()
+                .SetStateJournalText("Return to Sharene in the Hutlar Outpost and report your progress.")
+
+                .AddState()
+                .SetStateJournalText("Sharene wants you to kill another One Hundred Byysk. Better get going")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs:171]: More grammar stuff. "One Hundred" shouldn't be capitalized. 

"Sharene wants you to kill another one hundred Byysk. Better get going!"

File: SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs [hunk 4]
```diff
@@ -155,5 +156,42 @@ private void StupendousSlugBile()
...
+                .AddKillObjective(NPCGroupType.Byysk_Guardian, 100)
+
+                .AddState()
+                .SetStateJournalText("Return to Sharene, once again, at the Hutlar Outpost.")
+
+                .AddState()
+                .SetStateJournalText("Sharene wants you to kill ANOTHER One Hundred Byysk...")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs:178]: I'm already annoyed that you're making the player go kill 400 Byysk and I'm not even the one playing. We need to rethink this quest.

File: SWLOR.Game.Server/Feature/QuestDefinition/TatooineQuestDefinition.cs [hunk 5]
```diff
@@ -43,5 +44,42 @@ private static void StinkyWomprats(QuestBuilder builder)
                 .AddGoldReward(1000)
                 .AddXPReward(1750);
         }
+
+        private static void TuskenRampage(QuestBuilder builder)
+        {
+            builder.Create("tusken_rampage", "Tusken Rampage")
+
+                .AddState()
+                .SetStateJournalText("The Militia wants you to kill Fifty Tusken Raider's.")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/TatooineQuestDefinition.cs:53]: This is another one that's just going to piss off the players. Rethink this one too. If you want to make the target a high number that's fine, but don't make them come back for multiple stages.

Comment (zunath) [reply SWLOR.Game.Server/Feature/QuestDefinition/TatooineQuestDefinition.cs:53]: Also fix the grammar please. Run it through ChatGPT or something if you need to.

File: SWLOR.Game.Server/Feature/QuestDefinition/ViscaraQuestDefinition.cs [hunk 6]
```diff
@@ -542,5 +544,108 @@ private void JoiningTheRepublic()
...
+                .AddXPReward(2500)
+                .AddItemReward("recipe_fabmedic1", 1);
+        }
+
+        private void SpiceOneSmallFavour()
+        {
+            _builder.Create("spice_onesmallfavour", "Spice: One Small Favour")
+
+                .AddState()
+                .SetStateJournalText("Stephen needs you to gather some different fiberplast so he can make some cool new rags he saw in a Magazine.")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/ViscaraQuestDefinition.cs:570]: Don't need to capitalize "magazine".

File: SWLOR.Game.Server/Feature/QuestDefinition/ViscaraQuestDefinition.cs [hunk 7]
```diff
@@ -542,5 +544,108 @@ private void JoiningTheRepublic()
...
+
+                .AddState()
+                .SetStateJournalText("Now Stephen needs you to get him some electronics so that he can make a sick new electric guitar.")
+                .AddCollectItemObjective("elec_ruined", 12)
+                .AddCollectItemObjective("elec_flawed", 19)
+                .AddCollectItemObjective("elec_good", 24)
+                .AddCollectItemObjective("elec_imperfect", 17)
+
+                .AddState()
+                .SetStateJournalText("You gave him the electronics, but not sure if he even has power down here.")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/ViscaraQuestDefinition.cs:587]: Adjust for some clarity: "You gave him the electronics but you're not sure if he even has power down here."

File: SWLOR.Game.Server/Feature/QuestDefinition/ViscaraQuestDefinition.cs [hunk 8]
```diff
@@ -542,5 +544,108 @@ private void JoiningTheRepublic()
...
+                .AddCollectItemObjective("herb_v", 25)
+                .AddCollectItemObjective("herb_c", 14)
+                .AddCollectItemObjective("herb_t", 21)
+                .AddCollectItemObjective("herb_x", 36)
+
+                .AddState()
+                .SetStateJournalText("Not sure that's what he's going to use the herbs for, but who are you to judge?")
+
+                .AddState()
+                .SetStateJournalText("This is going to far now. Apparently his 'home' needs some decorations. Grab these creature pieces and see what happens.")
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/QuestDefinition/ViscaraQuestDefinition.cs:630]: to -> too

File: SWLOR.Game.Server/Feature/RecipeDefinition/FabricationRecipeDefinition/FurnitureRecipes.cs [hunk 9]
```diff
@@ -1674,20 +2052,43 @@ private void Tier3()
                 .Component("ancient_wood", 6)
                 .Component("lth_good", 3);
 
-            // DNA Extractor III
-            _builder.Create(RecipeType.DNAExtractor3, SkillType.Fabrication)
```
Comment (zunath) [inline SWLOR.Game.Server/Feature/RecipeDefinition/FabricationRecipeDefinition/FurnitureRecipes.cs:1678]: Looks like the DNA extractor got deleted in your copy. Make sure you re-add this to your copy.

Comment (NZNightwolf) [reply SWLOR.Game.Server/Feature/RecipeDefinition/FabricationRecipeDefinition/FurnitureRecipes.cs:1678]: It's down in line 2078

Tried to keep all the DNA extractors at the end of their respected tier for future proofing.

Comment (zunath) [reply SWLOR.Game.Server/Feature/RecipeDefinition/FabricationRecipeDefinition/FurnitureRecipes.cs:1678]: Ok, the recipes are in order of level generally speaking (might be a few exceptions) but that's fine. Just wanted to make sure it didn't get deleted. I see it now thanks

**Comment by NZNightwolf** [comment]
https://docs.google.com/spreadsheets/d/1mqX1mNnWQl2-a_Oe7NCi3iyc_3gSpD3aZDx4k7hH07c/edit?gid=262340382#gid=262340382

File: SWLOR.Game.Server/Feature/QuestDefinition/TatooineQuestDefinition.cs (hunk 5 above)
Comment (NZNightwolf) [reply SWLOR.Game.Server/Feature/QuestDefinition/TatooineQuestDefinition.cs:53]: Dropped this to a single stage of hundred and fifty tusken

File: SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs (hunk 4 above)
Comment (NZNightwolf) [reply SWLOR.Game.Server/Feature/QuestDefinition/HutlarQuestDefinition.cs:178]: Droped this to a single stage two hundred and fifty Byssk

File: Module/dlg/cz220_5cr4pconvo.dlg.json [hunk 10]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "<StartHighlight>[The droid looks at you with a glint in it's torch.]</Start>"
```
Comment (zunath) [inline Module/dlg/cz220_5cr4pconvo.dlg.json:351]: "it's" -> "its"

File: Module/dlg/cz220_5cr4pconvo.dlg.json [hunk 11]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "<StartHighlight>[The droid seems happy with it's carts filled with scrap metal.]</Start>"
```
Comment (zunath) [inline Module/dlg/cz220_5cr4pconvo.dlg.json:538]: "it's" -> "its"

File: Module/dlg/cz220_5cr4pconvo.dlg.json [hunk 12]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": "action"
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Alright, i'll be back shortly."
```
Comment (zunath) [inline Module/dlg/cz220_5cr4pconvo.dlg.json:828]: "i'll" -> "I'll"

File: Module/dlg/cz220_5cr4pconvo.dlg.json [hunk 13]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": "action"
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Here you go buddy."
```
Comment (zunath) [inline Module/dlg/cz220_5cr4pconvo.dlg.json:930]: "Here you go buddy." -> "Here you go, buddy."

File: Module/dlg/cz220_5cr4pconvo.dlg.json [hunk 14]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": ""
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "You're welcome little droid! Good luck with all that scrap."
```
Comment (zunath) [inline Module/dlg/cz220_5cr4pconvo.dlg.json:995]: "You're welcome little droid! Good luck with all that scrap" -> "You're welcome, little droid! Good luck with all that scrap."

File: Module/dlg/mon_p3dr0oilpit.dlg.json [hunk 15]
```diff
@@ -0,0 +1,1025 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Greetings! This unit would request that you, being infront of me, is to aid me. State your decision... Now."
```
Comment (zunath) [inline Module/dlg/mon_p3dr0oilpit.dlg.json:106]: "Greetings! This unit requests that you, being in front of me, aid me. State your decision now."

File: Module/dlg/mon_p3dr0oilpit.dlg.json [hunk 16]
```diff
@@ -0,0 +1,1025 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Outstanding! P3DR0 would request an establishment to, as the beings say, party. Please advise if you can gather the following equipment for me: Five Speakers, One Jukebox, and One schematic for Cantina."
```
Comment (zunath) [inline Module/dlg/mon_p3dr0oilpit.dlg.json:194]: "Outstanding! P3DR0 requests an establishment for what the beings refer to as a party. Please advise if you can gather the following equipment for me: five speakers, one jukebox, and one schematic for a cantina."

File: Module/dlg/mon_p3dr0oilpit.dlg.json [hunk 17]
```diff
@@ -0,0 +1,1025 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Greetings! This unit would like to inquire if the, being infront of me, has managed to obtain the requested items."
```
Comment (zunath) [inline Module/dlg/mon_p3dr0oilpit.dlg.json:282]: "Greetings! This unit would like to inquire if you, being in front of me, have managed to obtain the requested items."

File: Module/dlg/mon_p3dr0oilpit.dlg.json [hunk 18]
```diff
@@ -0,0 +1,1025 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Outstanding, being in front of me! This unit would like to congratulate you on assisting me."
```
Comment (zunath) [inline Module/dlg/mon_p3dr0oilpit.dlg.json:351]: "Outstanding, being in front of me! This unit would like to congratulate you, on assisting me."

File: Module/dlg/mon_p3dr0oilpit.dlg.json [hunk 19]
```diff
@@ -0,0 +1,1025 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "This unit certainly hopes so! I will be in contact once: The Oil Pit is open for being patronage."
```
Comment (zunath) [inline Module/dlg/mon_p3dr0oilpit.dlg.json:420]: "This unit certainly hopes so! I will be in contact once the Oil Pit is open for patronage."

This recommendation from ChatGPT is kind of a hit or miss, open to discuss.

File: Module/dlg/mon_p3dr0oilpit.dlg.json [hunk 20]
```diff
@@ -0,0 +1,1025 @@
...
+          "value": ""
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "I.. look forward to it?"
```
Comment (zunath) [inline Module/dlg/mon_p3dr0oilpit.dlg.json:918]: "I.." -> "I..."

File: Module/dlg/veles_shelbquest.dlg.json [hunk 21]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "If you could get me ten new Medical Bed's, and perhaps the schematics for a new medical center? That'd be great!"
```
Comment (zunath) [inline Module/dlg/veles_shelbquest.dlg.json:263]: "Bed's" -> "Beds"

File: Module/dlg/veles_shelbquest.dlg.json [hunk 22]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Thanks for your help again! If anything else pops up, i'll be sure to reach out."
```
Comment (zunath) [inline Module/dlg/veles_shelbquest.dlg.json:538]: "i'll" -> "I'll"

File: Module/dlg/veles_shelbquest.dlg.json [hunk 23]
```diff
@@ -0,0 +1,1159 @@
...
+          "value": ""
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Of course, i'll be right back."
```
Comment (zunath) [inline Module/dlg/veles_shelbquest.dlg.json:828]: "i'll" -> "I'll"

File: Module/dlg/visc_spicesteph.dlg.json [hunk 24]
```diff
@@ -0,0 +1,5514 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Oh... This will be -easy-. I need some rocks to hold the corners down so I can enjoy my picnic's... <StartAction>[He holds out the list... again.]</Start>"
```
Comment (zunath) [inline Module/dlg/visc_spicesteph.dlg.json:1319]: "picnic's" -> "picnics"

File: Module/dlg/visc_spicesteph.dlg.json [hunk 25]
```diff
@@ -0,0 +1,5514 @@
...
+          "value": ""
+        },
+        "Speaker": {
+          "type": "cexostring",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "There was these two guys in hood's a couple weeks ago, and they were talking about these shiny shards that can be found on some bigger creatures on Dathomir. I have no idea, but I think it'd look pretty cool hung up in the center of the room."
```
Comment (zunath) [inline Module/dlg/visc_spicesteph.dlg.json:2111]: "hood's" -> "hoods"

File: Module/dlg/visc_spicesteph.dlg.json [hunk 26]
```diff
@@ -0,0 +1,5514 @@
...
+          "value": ""
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Not excatly... I'll be back. "
```
Comment (zunath) [inline Module/dlg/visc_spicesteph.dlg.json:3300]: "excatly" -> "exactly"

File: Module/dlg/visc_spicesteph.dlg.json [hunk 27]
```diff
@@ -0,0 +1,5514 @@
...
+          "value": "action"
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "This wasn't to bad... Here you go."
```
Comment (zunath) [inline Module/dlg/visc_spicesteph.dlg.json:3357]: "to" -> "too"

File: Module/dlg/visc_spicesteph.dlg.json [hunk 28]
```diff
@@ -0,0 +1,5514 @@
...
+          "value": ""
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Rock's are harder to find then I though... I'll be back."
```
Comment (zunath) [inline Module/dlg/visc_spicesteph.dlg.json:3928]: "Rock's" -> "Rocks"

Comment (zunath) [inline Module/dlg/visc_spicesteph.dlg.json:3928]: "though" -> "thought"

File: Module/dlg/visc_spicesteph.dlg.json [hunk 29]
```diff
@@ -0,0 +1,5514 @@
...
+          "value": ""
+        },
+        "Sound": {
+          "type": "resref",
+          "value": ""
+        },
+        "Text": {
+          "type": "cexolocstring",
+          "value": {
+            "0": "Nto quite, I wanted to ask what these are actually for... But, nevermind. Be back soon."
```
Comment (zunath) [inline Module/dlg/visc_spicesteph.dlg.json:4242]: "Nto" -> "Not"

**Review by zunath — DISMISSED** [review]
Requesting changes. Please run all of your dialogues through ChatGPT before submitting in the future, I don't want to go through all of them individually again.

File: Module/dlg/mon_p3dr0oilpit.dlg.json (hunk 19 above)
Comment (NZNightwolf) [reply Module/dlg/mon_p3dr0oilpit.dlg.json:420]: Edited to:

This unit certainly hopes so! I will be in contact once the Oil Pit construction has been completed.

**Review by taelon875 — DISMISSED** [review]

**Pull request merged**